### PR TITLE
feat(rsync-port): Add support for non-standard ssh port for rsync backup

### PR DESCRIPTION
**Pull request by hughsw**
These changes add that the user can configure the ssh port used for rsync backup.  This was surprising simple in the end because the existing plumbing for the target url already implicitly supported the port number because URL syntax includes an option for the port in the net_loc slot: https://datatracker.ietf.org/doc/html/rfc1738#section-3.1

Un-intrusive UI change just adds guiding commentary to Hostname field
image

This PR addresses the core concerns of several Issues:
- #2104
- #1914 
- #1758 
- #1147 

See also PR #2207 which fixes the broken display of rsync settings (and where I realized that supporting port configuration isn't difficult).

### CODE REVIEW

**Comment by JoshData** [comment]
TY/

**Comment by schlypel** [comment]
Hi and thank you for the ability to set a port in rsync.

tl;dr critical bug if no port is given, default to "22" broken, comandline shows "None" instead of "22"

It appears the way you check for a given port and default to "22" if none set is not working.
I just updated to v61 where this is merged and received the following error:

```
root@box:~/mailinabox# ./management/backup.py 
Giving up after 5 attempts. BackendException: Error running 'rsync -e 'ssh  -oBatchMode=yes  -i /root/.ssh/id_rsa_miab -p None'  -e "/usr/bin/ssh -oStrictHostKeyChecking=no -oBatchMode=yes -p **None** -i /root/.ssh/id_rsa_miab" miab@A.B.C.D:/home/miab/encrypted/': returned 255, with output:

Bad port 'None'
rsync: connection unexpectedly closed (0 bytes received so far) [Receiver]
rsync error: unexplained error (code 255) at io.c(228) [Receiver=3.2.3]
```

I read up on your changes and tried setting ":22" after my target IP in the miab settings. That fixed it, but I suppose all rsync backups with not explicitly set port will be broken right now.

**Comment by hughsw** [comment]
@JoshData, @SConaway, I have made a PR #2226 that fixes this issue.

**Comment by SConaway** [comment]
@hughsw I think you @'ed the wrong person here...

**Comment by hughsw** [comment]
Whoops.  Yep.  Fixed it.
